### ISSUE TITLE
"Reset to Defaults" doesn't clear shaper pipe/queue

### ISSUE THREAD

**Issue by Manley1995** [opnsense/core#6080]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [ X] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [ X] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Describe the bug**

When I reset OPNsense to defaults from the GUI, the shaper pipes/queues from the prior configuration remain in effect.

**To Reproduce**

1. Configure system with working traffic shaper (Firewall/Shaper/pipes/queues/rules) and test with a common bufferbloat site (like Waveform's) to see that shaper is working properly
2. Then reset system to defaults:  Go to 'System: Configuration: Defaults'  Click on 'Are you sure you want to proceed? [YES]'  
3. System reboots to default settings 
4. Perform a new configuration so router is working, but don't set up the shaper.
5. Download the the config.xml file, open it with a text editor, and confirm that the shaper entries for pipes, queues and rules are all empty.
6. Test bufferbloat again to see that the test results are still good
7. Go to the menus in the GUI and look under Firewall/Shaper/Status, and see that the bandwidth settings from the original configuration are still there and the pipes/queues are still processing packets (the "i" icons in the bandwidth column say droptail is being used).  See screenshot attached.

**Expected behavior**

I expected the shaper rules to be fully deleted when I reset to defaults (and bufferbloat test results to be worse again).

**Describe alternatives you considered**

If I needed a different shaper config, I would have just entered it with the GUI.  
If I needed no shaper, I guess I'd do a clean install from scratch.

**Screenshots**

![image](https://user-images.githubusercontent.com/115550830/195199678-4b48e885-9689-4051-ae43-3c8e021f4bfa.png)

**Relevant log files**

If applicable, information from log files supporting your claim.

**Additional context**

APU4C4, 300/10 cable connection.  Prior shaper config followed instructions from https://maltechx.de/en/2021/03/opnsense-setup-traffic-shaping-and-reduce-bufferbloat/ 

Software version used and hardware type if relevant, e.g.:

OPNsense 22.7.5 (amd64, OpenSSL).
PC Engines APU4C4
Network Intel I211

**Comment by AdSchellevis** [comment]
That does sound a bit unlikely, a reboot flushes the config files after which ipfw won't/can't start anymore. 

Things you could take a look at:

rc (start) config:
```
cat /etc/rc.conf.d/ipfw 
```

rules file
```
cat /usr/local/etc/ipfw.rules
```

And pipes defined in the config.xml file.
```
grep '<pipe' /conf/config.xml
```

If these are populated, the following command writes the templates to disk, which always happens on boot as well:
`configctl template reload OPNsense/IPFW`

(https://github.com/opnsense/core/blob/master/src/etc/rc.syshook.d/early/15-templates)

**Comment by fichtner** [comment]
The forum post https://forum.opnsense.org/index.php?topic=30630.0

**Comment by AdSchellevis** [comment]
In that case the relevant question probably is if the machine did shutdown after pressing the button, given the actions performed by the page.

https://github.com/opnsense/core/blob/26ccc7839ff4911970c8d340a3e61b41a0740f0c/src/etc/inc/system.inc#L1194-L1203

https://github.com/opnsense/core/blob/af587e812a7b8d96cf3fd84869dc7e00bbfe09b3/src/www/diag_defaults.php#L100-L106

On my end this all looks normal (page does flush config and powers the machine down).

**Comment by Manley1995** [comment]
I tried two experiments.

First experiment - I downloaded the config.xml file via the GUI, confirmed that the shaper entries were empty.

    TrafficShaper version="1.0.3"
      pipes/
      queues/
      rules/
    /TrafficShaper

I then immediately restored that same config.xml (which was followed by an automatic reboot).  The Firewall/Shaper/Status still showed the 310 and 11 Mbps entries as shown in the earlier screen shot.

Second experiment -  I did a reset to defaults (via the GUI), and allowed the APU to fully power down (front panel power LED off).  I unplugged the power supply from the wall, moved my LAN ethernet cable to OPNsense's default port 0 (I normally have it on port 1), and plugged the power supply back into the wall.  The APU booted, I logged in, did the minimal configuration, and checked the Firewall/Shaper/Status page.  I see the same BW settings as before (310/11).  Here's the screenshot taken minutes ago after doing a factory reset, power cycle, and nothing more:
 
![image](https://user-images.githubusercontent.com/115550830/195461897-a0c99081-a2a6-49e9-9016-0688d3d6e230.png)

Then I restored the config.xml with all my normal settings (but no shaper), and I am still seeing the shaper status page with the 310/11 Mbps settings.  (I also checked to see if the BW settings were somewhere else in the config.xml file - the digits '310' do not appear anywhere in the file.)

So basically no change despite the reset to defaults and full power cycle.

**Comment by AdSchellevis** [comment]
ok, but what do these files look like after factory reset and reboot?
```
cat /etc/rc.conf.d/ipfw 
```

```
cat /usr/local/etc/ipfw.rules
```

Just to confirm, the settings page doesn't show any pipes/queues/rules after the reboot, right?

**Comment by fichtner** [comment]
TBH: Looks like browser playing tricks, try a different one or reset browser cache.


Cheers,
Franco

**Comment by Manley1995** [comment]
It looks like it's in the ipfw.rules file.

root@OPNsense:~ # cat /usr/local/etc/ipfw.rules

#======================================================================================
# flush ruleset
#======================================================================================
flush

#======================================================================================
# define dummynet pipes
#======================================================================================
pipe 10000 config bw 310000Kbit/s queue 2 type  fq_codel  ecn    quantum 600   
pipe 10001 config bw 11000Kbit/s type  fq_codel  ecn     

#======================================================================================
# define dummynet queues
#======================================================================================
queue 10000 config pipe 10000 mask dst-ip 0xffffffff  weight 100
queue 10001 config pipe 10001 mask src-ip 0xffffffff  weight 100


#======================================================================================
# general purpose rules 1...1000
#======================================================================================
add 100 allow pfsync from any to any
add 110 allow carp from any to any
# layer 2: pass ARP
add 120 pass layer2 mac-type arp,rarp
# OPNsense requires for WPA
add 130 pass layer2 mac-type 0x888e,0x88c7
# PPP Over Ethernet Session Stage/Discovery Stage
add 140 pass layer2 mac-type 0x8863,0x8864
# layer 2: block anything else non-IP(v4/v6)
add 150 deny layer2 not mac-type ip,ipv6

# allow traffic send from localhost
add 200 skipto 60000 ipv6 from ::1 to any
add 201 skipto 60000 ipv4 from 127.0.0.0/8 to any
add 202 skipto 60000 ipv6 from any to ::1
add 203 skipto 60000 ipv4 from any to 127.0.0.0/8

#======================================================================================
# Allow traffic to this host
#======================================================================================



#======================================================================================
# redirect non-authenticated clients to captive portal @ local port 8000 + zoneid
#======================================================================================


#======================================================================================
# accept traffic from all interfaces not used by captive portal
#======================================================================================
# let the responses from the captive portal web server back out
add 6000 skipto 60000 tcp from any to any out
# forward unauthorized traffic from captiveportal interfaces to block rule
# send all the rest to the traffic shaper rules
add 6199 skipto 60000 all from any to any


#======================================================================================
# 30000 .... 49999 reserved for captive portal accounting rules
#======================================================================================


#======================================================================================
# traffic shaping section, authorized traffic
#======================================================================================
add 60000 return via any

add 60001 queue 10000 ip from any to any src-port  any dst-port any in   via igb0 // b0e55858-e486-4a4b-9de7-c78616937fd5 wan: Download Queue
add 60002 queue 10001 ip from any to any src-port  any dst-port any out   via igb0 // 80421202-1caa-4787-a05a-58356d958050 wan: Upload Queue



# pass authorized
add 65533 pass ip from any to any

# block all unmatched
add 65534 deny all from any to any
root@OPNsense:~ #

**Comment by AdSchellevis** [comment]
and after executing the following command?

```
configctl template reload OPNsense/IPFW
```

**Comment by Manley1995** [comment]
That appears to have cleared it.

root@OPNsense:~ # configctl template reload OPNsense/IPFW
OK
root@OPNsense:~ # cat /usr/local/etc/ipfw.rules

#======================================================================================
# flush ruleset
#======================================================================================
flush

#======================================================================================
# define dummynet pipes
#======================================================================================

#======================================================================================
# define dummynet queues
#======================================================================================

---rest of file---

**Comment by Manley1995** [comment]
Sorry about the formatting (am new to this).

**Comment by Manley1995** [comment]
Yes, after a reboot, the Firewall/Shaper/Status page no longer shows the pipes/queue/rules.  That was it.

**Comment by fichtner** [comment]
@Manley1995 nice, close this issue then?

**Comment by Manley1995** [comment]
I don't think we can close it quite yet.  While "configctl template reload OPNsense/IPFW" did reset what was in ipfw.rules, it didn't address the fact that ipfw.rules wasn't originally reset by the "Reset to Defaults" process.  After manually resetting the ipfw.rules (that last step above), I did another experiment.  

1. I looked at Firewall/Shaper/Status after having reset the ipfw.rules file.  I see this:
![Shaper Status No Shaper](https://user-images.githubusercontent.com/115550830/195920469-00cc0536-5669-434f-8b4d-297e4e3c11e6.jpeg)
2. I reconfigured the shaper rules (using that maltechx.de site I mentioned above)
3. I checked to see if the new shaper showed up in Firewall/Shaper/Status - it did
![Shaper Status with Shaper (2)](https://user-images.githubusercontent.com/115550830/195921262-56f1d8eb-acf7-4b6d-84b4-3bd5043b9a27.jpeg)
4. I tested bufferbloat using Waveform's site (superb results, by the way)
5. I again Reset to Defaults from the GUI, full power down, unplug from the wall
6. I deleted my browser's cache (everything for all time, just in case)
7. I moved my LAN cable (to OPNsense's default port 0) and plugged the power back in to boot
8. I logged in, did just enough configurations to get to the dashboard, and looked at the Firewall/Shaper/Status.  The shaper pipes and queues are back again, left over from my latest shaper configuration (step 2 in this message).

![Shaper Status After Reset](https://user-images.githubusercontent.com/115550830/195915908-b60d95b7-57f4-489c-afad-571a99d46ee6.jpeg)

So I think there's a bug - the command that AdSchellevis suggested does reset the ipfw.rules, but the Reset-to-Defaults process apparently doesn't.  I've Reset-to-Defaults three times, and if the shaper status isn't empty before the reset, then the shaper status isn't empty after the reset. (Notice, in the Firewall/Shaper/Status screenshot, after a Reset to Defaults, the Descriptions are all missing - is that a clue?)

**Comment by AdSchellevis** [comment]
Reloading the templates is always executed on boot, using an early boot hook https://github.com/opnsense/core/blob/master/src/etc/rc.syshook.d/early/15-templates

If it didn't flush, this might help https://github.com/opnsense/core/commit/e2c182bd4b51ba59e6f6c1631b97238052a618e0
